agent/generator.go

```diff
@@ -44,7 +44,11 @@ func generateValues(generators []metrics.Generator) chan metrics.Values {
 					logger.Errorf("Failed to generate value in %T (skip this metric): %s", g, err.Error())
 					return
 				}
-				processed <- values
+				var customIdentifier *string
+				if pluginGenerator, ok := g.(metrics.PluginGenerator); ok {
+					customIdentifier = pluginGenerator.CustomIdentifier()
+				}
+				processed <- metrics.ValuesCustomIdentifier{values, customIdentifier}
```

Comment:
[nit] use `{Value: value, CustomIdentifier: customIdentifier}` format for forward compatibility.